TypeSystem/flow-analysis/reachability_while_A03_t01.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion while statement: If `N` is a while statement of the form
+/// `while (E) S` then:
+/// - Let `before(E) = conservativeJoin(before(N), assignedIn(N), capturedIn(N))`
+/// - Let `before(S) = split(true(E))`.
+/// - Let `after(N) = inheritTested(join(false(E), unsplit(break(S))), after(S))`
+///
+/// @description Checks that if `E` is not a boolean literal flow analysis
+/// detects assignment in `S`.
+/// @author sgrekhov22@gmail.com
+
+main() {
+  late int i;
+  if (1 > 2) {
+    while (true) {
+      if (false) {
+        break;
+      }
```

Comment:
Interesting! It looks right, we're joining the unreachable `false(E)` and the unreachable `break(S)` and obtaining an `after(N)` which is also unreachable.

I hadn't expected the flow analysis to be so precise, but it looks like it should work. ;-)